crates/matrix-sdk/src/authentication/qrcode/secure_channel.rs

```diff
@@ -50,7 +50,7 @@ impl SecureChannel {
     pub(super) async fn new(http_client: HttpClient, homeserver_url: &Url) -> Result<Self, Error> {
         let channel = RendezvousChannel::create_outbound(http_client, homeserver_url).await?;
         let rendezvous_url = channel.rendezvous_url().to_owned();
-        let mode_data = QrCodeModeData::Reciprocate { homeserver_url: homeserver_url.clone() };
+        let mode_data = QrCodeModeData::Reciprocate { server_name: homeserver_url.to_string() };
```

Comment:
Maybe a missed renaming for the `homeserver_url` parameter?

Reply:
This one actually isn't, we pass in the resolved homeserver URL for this test method. Kinda abusing the API, but fine for now.

Reply:
Added a comment explaining this.